src/transformers/models/speech_to_text/modeling_tf_speech_to_text.py

```diff
@@ -143,7 +143,9 @@ def __init__(self, config: Speech2TextConfig, **kwargs):
         ]
 
     def call(self, input_features: tf.Tensor) -> tf.Tensor:
-        hidden_states = tf.identity(input_features)  # TF Conv1D assumes Batch x Time x Channels, same as the input
+        hidden_states = tf.cast(
+            input_features, tf.float32
+        )  # TF Conv1D assumes Batch x Time x Channels, same as the input
```

Comment:
I'm not sure what the `tf.identity` was doing here - will removing this break anything @gante?

Casting to float to enable the `tf.concat` in graph mode in L156

Reply:
If I remember correctly, this line only existed to rename the variable (and probably `hidden_states = input_features` would be enough for that 🤔 )

Reply:
Could it be a mutation thing? tf.identity returns a copy

Reply:
To be frank I can't remember if that was the cause :D But since it was my first HF PR, I would err towards a bad choice on my end